.github/workflows/deploy.yml

```diff
@@ -4,11 +4,82 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
+env:
+ DOCKERHUB_USERNAME: 'dfedigital'
+ DOCKERHUB_IMAGE_NAME: 'teaching-vacancies'
 
 jobs:
-  deploy:
-    name: Deploy to Production
+  build_image:
+    name: build & publish docker image
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout Code
+
+    - name: prepare env
+      run: |
+        gpg --quiet --batch --yes --decrypt --passphrase="$DOTENV_PASSPHRASE" --output .env dotenv.gpg
+      env:
+        DOTENV_PASSPHRASE: ${{ secrets.DOTENV_PASSPHRASE }}
+
+    - name: docker build
+      run: |
+        make build-production
 
+    - name: docker push
+      run: |
+        docker login -u vigneshndfe -p $DOCKERHUB_PASSWORD
+        docker images
+        docker push $DOCKERHUB_USERNAME/$DOCKERHUB_IMAGE_NAME:$GITHUB_SHA
+        docker logout
+      env:
+        DOCKERHUB_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+  deploy_image:
+    needs: [build_image]
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
The team deploys to the dev environment manually to test integrations. Let's just deploy for staging for now?